twitter_api/main.py

```diff
@@ -1,28 +1,278 @@
 import sqlite3
-
-from flask import Flask
+import json
+from flask import Flask, Response, request, abort
 from flask import g
 
-app = Flask(__name__)
+# extra modules we added
+from hashlib import md5
+import random
+from .utils import JSON_MIME_TYPE, auth_only, json_only
+import string
+from datetime import datetime
 
+app = Flask(__name__)
+#api = Api(app)
 
 def connect_db(db_name):
     return sqlite3.connect(db_name)
 
-
 @app.before_request
 def before_request():
     g.db = connect_db(app.config['DATABASE'])
+    users = g.db.execute('SELECT access_token FROM auth')
+    
+# implement your views here
+@app.route('/login', methods = ['POST'])
+@json_only
+def login():
+    data = request.get_json()
+    try:
+        post_username = data['username']
+        post_password = data['password']
+    except KeyError:
+        abort(400) # Bad request
+    
+    sql_command = 'SELECT username, password, id FROM user WHERE username = ?'
+    
+    found_user_cursor = g.db.execute(sql_command, [post_username])
+    found_user = found_user_cursor.fetchone()
+    try:
+        username, password, id = found_user
+    except TypeError: # if username did find a match in db
+        abort(404) # Not found
+        
+    post_password = post_password.encode('utf-8') # needs to be in bytes
```

Comment:
The request can be encoded with a different charset than `UTF-8`. If it's coming from Japan, it's JIS for example. It's better to do: `post_password.encode(request.charset)`.